src/main/java/me/botsko/prism/wands/InspectorWand.java

```diff
@@ -1,137 +1,118 @@
-				if (results.getActionResults().size() > 5) {
-					player.sendMessage(Prism.messenger.playerHeaderMsg("Showing " + results.getTotalResults()
-							+ " results. Page 1 of " + results.getTotal_pages()));
-				}
-				for (final me.botsko.prism.actions.Handler a : results.getPaginatedActionResults()) {
-					final ActionMessage am = new ActionMessage(a);
-					if (parameters.hasFlag(Flag.EXTENDED)
-							|| plugin.getConfig().getBoolean("prism.messenger.always-show-extended")) {
-						am.showExtended();
-					}
-					player.sendMessage(Prism.messenger.playerMsg(am.getMessage()));
-				}
-			} else {
-				final String space_name = (block.getType().equals(Material.AIR) ? "space"
-						: block.getType().toString().replaceAll("_", " ").toLowerCase()
-						+ (block.getType().toString().endsWith("BLOCK") ? "" : " block"));
-				player.sendMessage(Prism.messenger.playerError("No history for this " + space_name + " found."));
-			}
-		});
-	}
+public class InspectorWand extends QueryWandBase {
 
-	/**
-	 * 
-	 */
-	@Override
-	public void playerRightClick(Player player, Entity entity) {
-	}
+    /**
+     * @param plugin
+     */
+    public InspectorWand(Prism plugin) {
+        super(plugin);
+    }
+
+    /**
+     *
+     */
+    @Override
+    public void playerLeftClick(Player player, Location loc) {
+        showLocationHistory(player, loc);
+    }
+
+    /**
+     *
+     */
+    @Override
+    public void playerRightClick(Player player, Location loc) {
+        showLocationHistory(player, loc);
+    }
+
+    /**
+     * Show the Location History for a player.
+     *
+     * @param player Player
+     * @param loc    Location
+     */
+    protected void showLocationHistory(final Player player, final Location loc) {
+
+        final Block block = loc.getBlock();
+        final Block sibling = BlockUtils.getSiblingForDoubleLengthBlock(block);
+        plugin.getServer().getScheduler().runTaskAsynchronously(plugin, () -> {
+
+            // Build params
+            QueryParameters params;
+
+            try {
+                params = parameters.clone();
+            } catch (final CloneNotSupportedException ex) {
+                params = new QueryParameters();
+                player.sendMessage(Prism.messenger
+                        .playerError("Error retrieving parameters. Checking with default parameters."));
+            }
+            params.setWorld(player.getWorld().getName());
+            params.setSpecificBlockLocation(loc);
+
+            // Do we need a second location? (For beds, doors, etc)
+            if (sibling != null) {
+                params.addSpecificBlockLocation(sibling.getLocation());
+            }
+
+            // Ignoring any actions via config?
+            if (params.getActionTypes().size() == 0) {
+                @SuppressWarnings("unchecked") final Collection<String> ignoreActions = (ArrayList<String>) plugin.getConfig()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line is longer than 120 characters (found 126).](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5438808)